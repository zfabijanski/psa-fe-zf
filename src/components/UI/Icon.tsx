import { FC, SVGProps } from "react";
import styled from "styled-components/macro";
import { TDefaultThemeColor } from "../../theme/theme";

import { ReactComponent as AlertFill } from "../../assets/newIcons/alert-fill.svg";
import { ReactComponent as AlertTriangle } from "../../assets/newIcons/alert-triangle.svg";
import { ReactComponent as AlertDoubleTriangle } from "../../assets/newIcons/alert-double-triangle.svg";
import { ReactComponent as IllustrationFrameRounded } from "../../assets/newIcons/illustration-frame-rounded.svg";
import { ReactComponent as ArrowRight } from "../../assets/newIcons/arrow-right.svg";
import { ReactComponent as ArrowLeft } from "../../assets/newIcons/arrow-left.svg";
import { ReactComponent as ArrowUp } from "../../assets/newIcons/arrow-up.svg";
import { ReactComponent as Calculator } from "../../assets/newIcons/calculator.svg";
import { ReactComponent as CalculatorPlus } from "../../assets/newIcons/calculator-plus.svg";
import { ReactComponent as Calendar } from "../../assets/newIcons/calendar.svg";
import { ReactComponent as CornerDownLeft } from "../../assets/newIcons/corner-down-left.svg";
import { ReactComponent as Check } from "../../assets/newIcons/check.svg";
import { ReactComponent as ChevronDown } from "../../assets/newIcons/chevron-down.svg";
import { ReactComponent as ChevronDown2 } from "../../assets/newIcons/chevron-down-2.svg";
import { ReactComponent as ChevronDown3 } from "../../assets/newIcons/chevron-down-3.svg";
import { ReactComponent as ChevronLeft } from "../../assets/newIcons/chevron-left.svg";
import { ReactComponent as ChevronRight } from "../../assets/newIcons/chevron-right.svg";
import { ReactComponent as ChevronUp } from "../../assets/newIcons/chevron-up.svg";
import { ReactComponent as FileText } from "../../assets/newIcons/file-text.svg";
import { ReactComponent as Info } from "../../assets/newIcons/info.svg";
import { ReactComponent as Logout } from "../../assets/newIcons/logout.svg";
import { ReactComponent as Padlock } from "../../assets/newIcons/padlock.svg";
import { ReactComponent as Plus } from "../../assets/newIcons/plus.svg";
import { ReactComponent as Search } from "../../assets/newIcons/search.svg";
import { ReactComponent as Trash2 } from "../../assets/newIcons/trash-2.svg";
import { ReactComponent as Trash } from "../../assets/newIcons/trash.svg";
import { ReactComponent as Refresh } from "../../assets/newIcons/refresh.svg";
import { ReactComponent as UserPlus } from "../../assets/newIcons/user-plus.svg";
import { ReactComponent as X } from "../../assets/newIcons/x.svg";
import { ReactComponent as Home } from "../../assets/newIcons/home.svg";
import { ReactComponent as Mail } from "../../assets/newIcons/mail.svg";
import { ReactComponent as MailX } from "../../assets/newIcons/mail-x.svg";
import { ReactComponent as Library } from "../../assets/newIcons/library.svg";
import { ReactComponent as Group } from "../../assets/newIcons/group.svg";
import { ReactComponent as Handshake } from "../../assets/newIcons/handshake.svg";
import { ReactComponent as People } from "../../assets/newIcons/people.svg";
import { ReactComponent as PruLogo } from "../../assets/newIcons/logo.svg";
import { ReactComponent as CommissionSystem } from "../../assets/newIcons/commission-system.svg";
import { ReactComponent as ElectronicApplication } from "../../assets/newIcons/electronic-application.svg";
import { ReactComponent as Products } from "../../assets/newIcons/products.svg";

import { ReactComponent as OldCheckWhite } from "../../assets/icons/check-white.svg";

const mapping = {
  "alert-fill": <AlertFill />,
  "alert-triangle": <AlertTriangle />,
  "alert-double-triangle": <AlertDoubleTriangle />,
  "illustration-frame-rounded": <IllustrationFrameRounded />,
  "arrow-right": <ArrowRight />,
  "arrow-left": <ArrowLeft />,
  "arrow-up": <ArrowUp />,
  calculator: <Calculator />,
  "calculator-plus": <CalculatorPlus />,
  calendar: <Calendar />,
  check: <Check />,
  "corner-downn-left": <CornerDownLeft />,
  "chevron-down": <ChevronDown />,
  "chevron-down-2": <ChevronDown2 />,
  "chevron-down-3": <ChevronDown3 />,
  "chevron-left": <ChevronLeft />,
  "chevron-right": <ChevronRight />,
  "chevron-up": <ChevronUp />,
  "file-text": <FileText />,
  info: <Info />,
  logout: <Logout />,
  plus: <Plus />,
  refresh: <Refresh />,
  search: <Search />,
  "trash-2": <Trash2 />,
  trash: <Trash />,
  "user-plus": <UserPlus />,
  x: <X />,
  home: <Home />,
  mail: <Mail />,
  "mail-x": <MailX />,
  library: <Library />,
  group: <Group />,
  handshake: <Handshake />,
  people: <People />,
  padlock: <Padlock />,
  "pru-logo": <PruLogo />,
  "commission-system": <CommissionSystem />,
  "electronic-application": <ElectronicApplication />,
  products: <Products />,

  "old-check-white": <OldCheckWhite />,
};

export type TIconType = keyof typeof mapping;

export interface IIconProps extends Omit<SVGProps<SVGSVGElement>, "onClick"> {
  name: TIconType;
  width?: number;
  className?: string;
  onClick?: React.MouseEventHandler<HTMLDivElement>;
  color?: TDefaultThemeColor;
  "data-testid"?: string;
}

export const Icon: FC<IIconProps> = (props): JSX.Element | null => {
  const { name, width, className, onClick, color } = props;
  // TODO: fix broken loading of icons
  // const { name, width, className, onClick, color, ref } = props;
  // const ImportedIconRef = useRef<FC<SVGProps<SVGSVGElement>>>();
  // const [loading, setLoading] = useState(false);

  // useEffect(() => {
  //   let isComponentMounted = true;
  //   setLoading(true);

  //   (async (): Promise<void> => {
  //     try {
  //       const { default: namedImport } = await import(
  //         `!!@svgr/webpack?-svgo,+titleProp,+ref!../../assets/newIcons/${name}.svg`
  //       );
  //       ImportedIconRef.current = namedImport;
  //     } catch (err) {
  //       // throw err;
  //     } finally {
  //       isComponentMounted && setLoading(false);
  //     }
  //   })();

  //   return () => {
  //     isComponentMounted = false;
  //   };
  // }, [name]);

  // if (loading || !ImportedIconRef.current) return null;

  // const { current: ImportedIcon } = ImportedIconRef;

  return (
    <IconWrapper
      iconWidth={width}
      className={className}
      onClick={onClick}
      color={color}
      data-testid={props["data-testid"]}
    >
      {mapping[name as keyof typeof mapping]}
    </IconWrapper>
  );
};

export const setIconColor = (color: string, isImportant: boolean = false) => `
  svg g {
    &, path, circle, rect {
      stroke: ${color}${isImportant ? " !important" : ""};
    }

    rect {
      fill: ${color}${isImportant ? " !important" : ""};
    }

    tspan {
      stroke: transparent${isImportant ? " !important" : ""};
      fill: ${color}${isImportant ? " !important" : ""};
    }
  }

  svg > path {
    stroke: ${color}${isImportant ? " !important" : ""};
  }
`;

const IconWrapper = styled.div<
  Omit<IIconProps, "width" | "name"> & { iconWidth: number | undefined }
>`
  display: inline-block;
  cursor: ${({ onClick }) => (!!onClick ? "pointer" : "inherit")};

  > svg {
    ${({ iconWidth }) => iconWidth !== undefined && `width: ${iconWidth}px;`}
    height: 100%;
    display: block;
  }

  ${({ color, theme }) => color && setIconColor(theme.newColors[color], true)};
`;
