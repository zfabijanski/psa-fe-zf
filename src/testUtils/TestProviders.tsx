import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { Normalize } from "styled-normalize";
import GlobalStyles from "theme/globalStyles";
import { theme } from "theme/theme";
import rootStore, { configureStore, rootReducer as reducer } from "AppStore";
import { BrowserRouter } from "react-router-dom";
import { Modal } from "components/UI/Modal";
import { IntlProvider } from "react-intl";
import { plMessages } from "translations/pl";
import I18nProvider from "I18nProvider";

interface TestProvidersProps {
  store?: typeof rootStore;
  renderModal?: boolean;
  realIntlProvider?: boolean;
}

export const addTestProviders =
  ({
    store = configureStore({ reducer }),
    renderModal = true,
    // @TODO: update tests to use realIntlProvider everywhere
    realIntlProvider = false,
  }: TestProvidersProps): FC<PropsWithChildren<{}>> =>
  ({ children }) => {
    const TranslationsProvider = realIntlProvider ? IntlProvider : I18nProvider;

    return (
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <TranslationsProvider locale="pl" messages={plMessages}>
              <Normalize />
              <GlobalStyles />
              {children}
              {renderModal && <Modal />}
            </TranslationsProvider>
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

const customRender = (
  ui: ReactElement,
  options: Omit<RenderOptions, "options"> = {},
  testProviderOptions: TestProvidersProps = {}
) => render(ui, { wrapper: addTestProviders(testProviderOptions), ...options });

export * from "@testing-library/react";

// override render method
export { customRender as render };
