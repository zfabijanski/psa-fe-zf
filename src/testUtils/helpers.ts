export const sleep = (ms: number = 100): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));
