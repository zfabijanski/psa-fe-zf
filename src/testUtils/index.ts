export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
