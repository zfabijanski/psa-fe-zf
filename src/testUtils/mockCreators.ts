import {
  ICalculatorCoversConfig,
  IProductCovers,
} from "features/Calculator/types";
import {
  DocumentState,
  IDirectory,
  IDirectoryMetadata,
  IDocumentMetadata,
} from "features/Library/types";
import { IMeeting } from "features/Meeting/History/types";
import { IIllustration, ProductGuid } from "features/Products/types";
import { IModalData, ModalTypes } from "models/Modal";
import {
  Frequency,
  IFundRiskProfileConfig,
  IProductConfig,
  IProductFund,
  IProductGuarantee,
  InvestmentProfile,
  Role,
} from "models/common";
import {
  CalculationStatus,
  Direction,
  ICalculationDetailsDto,
  ICover,
  IInsured,
  YesNo,
} from "models/calculator";

export const createCalculationDetailsDto = (
  overrides?: Partial<ICalculationDetailsDto>
): ICalculationDetailsDto => ({
  calculation_id: 1,
  working_date: "2020-01-01",
  premiums_total: 100,
  guaranteed_death_benefit: 100,
  guaranteed_maturity_benefit: 100,
  projected_maturity_value_egr_m: 100,
  product_categories: [],
  uuid: "uuid",
  version: "version",
  source_system: "source_system",
  product_guid: 1,
  insureds_list: [],
  covers_list: [],
  funds_list: [],
  ...overrides,
});

export const createInsured = (overrides?: Partial<IInsured>): IInsured => ({
  insured_idx: 1,
  insured_uuid: "insured_uuid",
  age_in_months: 1,
  date_of_birth: "2020-01-01",
  role: Role.MAIN,
  ...overrides,
});

export const createCover = (overrides?: Partial<ICover>): ICover => ({
  cover_id: 1,
  cover_idx: 1,
  insured_idx: 1,
  code: "code",
  frequency_code: Frequency.Annually,
  payments: 1,
  direction_of_calculation: Direction.SumAssuredToPremium,
  investment_profile: InvestmentProfile.Balanced,
  indexation: YesNo.Yes,
  sum_assured: 1,
  premium_per_freq: 1,
  duration: 1,
  idx_option: YesNo.Yes,
  rop_option: YesNo.Yes,
  add_sum_assured: 1,
  ending_age: 1,
  exp_mat_ben: 1,
  ...overrides,
});

export const createCalculatorCoversConfig = (
  overrides?: Partial<ICalculatorCoversConfig>
): ICalculatorCoversConfig => ({
  coverName: "coverName",
  isChild: false,
  isHospital: false,
  ...overrides,
});

export const createCalculatorCoversConfigMap = (
  overrides?: Partial<Record<string, ICalculatorCoversConfig>>
): Record<string, ICalculatorCoversConfig> => ({
  coverName1: createCalculatorCoversConfig({ coverName: "coverName1" }),
  coverName2: createCalculatorCoversConfig({ coverName: "coverName2" }),
  ...overrides,
});

export const createProductCovers = (
  overrides?: Partial<IProductCovers>
): IProductCovers => ({
  mainCover: "mainCover",
  additionalCovers: ["additionalCover1", "additionalCover2"],
  wopCovers: ["wopCover1", "wopCover2"],
  childCovers: ["childCover1", "childCover2"],
  ...overrides,
});

export const createProductCoversMap = (
  overrides?: Partial<Record<string, IProductCovers>>
): Record<string, IProductCovers> => ({
  mainCover1: createProductCovers({ mainCover: "mainCover1" }),
  mainCover2: createProductCovers({ mainCover: "mainCover2" }),
  ...overrides,
});

export const createProductConfig = (
  overrides?: Partial<IProductConfig>
): IProductConfig => ({
  product_guid: 1,
  product_name: "product_name",
  product_categories: [],
  product_freqs: [],
  product_covers: [],
  product_funds: [],
  ...overrides,
});

export const createProductConfigMap = (
  overrides?: Partial<Record<string, IProductConfig>>
): Record<string, IProductConfig> => ({
  1: createProductConfig({ product_name: "product_name1" }),
  2: createProductConfig({ product_name: "product_name2" }),
  ...overrides,
});

export const createFundRiskProfileConfig = (
  overrides?: Partial<IFundRiskProfileConfig>
): IFundRiskProfileConfig => ({
  code: "code",
  name_pl: "name_pl",
  name_en: "name_en",
  risk_level: 1,
  status_code: "status_code",
  notes: "notes",
  ...overrides,
});

export const createProductGuarantee = (
  overrides?: Partial<IProductGuarantee>
): IProductGuarantee => ({
  guarantee_code: "guarantee_code",
  guarantee_name: "guarantee_name",
  guarantee_min_value: "1",
  guarantee_duration_months: "4",
  guarantee_percentages: [{ guarantee_percent: 10 }],
  ...overrides,
});

export const createProductFund = (
  overrides?: Partial<IProductFund>
): IProductFund => ({
  cover_code: "cover_code",
  fund_code: "fund_code",
  fund_name: "fund_name",
  fund_risk_profile: "code",
  fund_order: 1,
  guarantees: [createProductGuarantee()],
  ...overrides,
});

export const createModalData = (
  overrides: Partial<IModalData> = {}
): IModalData => ({
  modalIllustration: "alert-fill",
  modalTitleTrKey: "modalTitleTrKey",
  modalContentTrKey: "modalContentTrKey",
  modalContent: "modalContent",
  modalButtons: {
    confirm: { textTrKey: "modal.ok" },
  },
  modalType: ModalTypes.default,
  maskClosable: true,
  ...overrides,
});

export const createIllustration = (
  overrides: Partial<IIllustration> = {}
): IIllustration => ({
  idd_id: 1,
  product_guid: ProductGuid.Savings,
  adequacy_id: 1,
  order_id: 1,
  expected_amount: { value: 3 },
  premium_total: { value: 3 },
  duration: { value: 3 },
  premium_main_cover: { value: 3 },
  frequency: { value: 3 },
  sum_of_premiums: { value: 3 },
  sum_assured: { value: 3 },
  investment_profile: { value: 3 },
  pru_calc_calculation_id: 1,
  status: CalculationStatus.Illustrated,
  json_needs_analysis: null,
  ...overrides,
});

export const createDirectory = (
  overrides?: Partial<IDirectory>
): IDirectory => ({
  directoryId: 1,
  directories: [],
  documentsIds: [1, 2, 3],
  ...overrides,
});

export const createDirectoryMetadata = (
  overrides?: Partial<IDirectoryMetadata>
): IDirectoryMetadata => ({
  directoryId: 1,
  name: "name",
  isOpen: false,
  ...overrides,
});

export const createDocumentMetadata = (
  overrides?: Partial<IDocumentMetadata>
): IDocumentMetadata => ({
  documentId: 1,
  guid: "guid",
  name: "name",
  state: DocumentState.NotAttached,
  isStateChanging: false,
  moduleName: "PRU_LIBRARY",
  ...overrides,
});

export const createMeeting = (overrides?: Partial<IMeeting>): IMeeting => ({
  meetingNo: "meetingNo",
  meetingId: 1,
  name: "name",
  age: 18,
  lastAccessDate: new Date("2023-01-01"),
  createDate: new Date("2023-01-01"),
  statusId: 1,
  statusName: "statusName",
  statusDescription: "statusDescription",
  state: "state",
  haveAdequacy: false,
  haveCalculation: false,
  haveIdd: false,
});
