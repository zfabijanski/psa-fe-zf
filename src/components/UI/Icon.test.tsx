import { Icon, IIconProps, TIconType } from "./Icon";
import { render, screen } from "testUtils";

const setup = (props: Partial<IIconProps> = {}) =>
  render(<Icon name="check" {...props} data-testid="icon" />);

// prettier-ignore
const iconNames = ['alert-fill', 'alert-triangle', 'alert-double-triangle', 'illustration-frame-rounded', 'arrow-right', 'arrow-up', 'calculator', 'calculator-plus', 'calendar', 'check', 'corner-downn-left', 'chevron-down', 'chevron-down-2', 'chevron-down-3', 'chevron-left', 'chevron-right', 'chevron-up', 'file-text', 'info', 'logout', 'plus', 'refresh', 'search', 'trash-2', 'trash', 'user-plus', 'x', 'home', 'mail', 'mail-x', 'library', 'group', 'handshake', 'people', 'padlock', 'pru-logo', 'commission-system', 'electronic-application', 'products', 'old-check-white'] as const

describe("Icon", () => {
  it.each(iconNames)("renders %s icon", (iconName: TIconType) => {
    setup({ name: iconName });
    expect(screen.getByTestId("icon")).toMatchSnapshot();
  });

  it("handles onClick", () => {
    const onClick = jest.fn();
    setup({ name: "check", onClick });
    screen.getByTestId("icon").click();
    expect(onClick).toHaveBeenCalled();
  });

  it("handles different sizes", () => {
    setup({ width: 2137 });
    expect(screen.getByTestId("icon")).toMatchSnapshot();
  });

  it("accepts custom color", () => {
    setup({ color: "primary100" });
    expect(screen.getByTestId("icon")).toMatchSnapshot();
  });
});
