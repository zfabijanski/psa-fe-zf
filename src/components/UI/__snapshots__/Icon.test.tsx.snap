// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon accepts custom color 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #055A60 !important;
}

.c0 svg g rect {
  fill: #055A60 !important;
}

.c0 svg g tspan {
  stroke: transparent !important;
  fill: #055A60 !important;
}

.c0 svg > path {
  stroke: #055A60 !important;
}

<div
  class="c0"
  color="primary100"
  data-testid="icon"
>
  <svg>
    check.svg
  </svg>
</div>
`;

exports[`Icon handles different sizes 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  width: 2137px;
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    check.svg
  </svg>
</div>
`;

exports[`Icon renders alert-double-triangle icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    alert-double-triangle.svg
  </svg>
</div>
`;

exports[`Icon renders alert-fill icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    alert-fill.svg
  </svg>
</div>
`;

exports[`Icon renders alert-triangle icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    alert-triangle.svg
  </svg>
</div>
`;

exports[`Icon renders arrow-right icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    arrow-right.svg
  </svg>
</div>
`;

exports[`Icon renders arrow-up icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    arrow-up.svg
  </svg>
</div>
`;

exports[`Icon renders calculator icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    calculator.svg
  </svg>
</div>
`;

exports[`Icon renders calculator-plus icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    calculator-plus.svg
  </svg>
</div>
`;

exports[`Icon renders calendar icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    calendar.svg
  </svg>
</div>
`;

exports[`Icon renders check icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    check.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-down icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-down.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-down-2 icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-down-2.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-down-3 icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-down-3.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-left icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-left.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-right icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-right.svg
  </svg>
</div>
`;

exports[`Icon renders chevron-up icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    chevron-up.svg
  </svg>
</div>
`;

exports[`Icon renders commission-system icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    commission-system.svg
  </svg>
</div>
`;

exports[`Icon renders corner-downn-left icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    corner-down-left.svg
  </svg>
</div>
`;

exports[`Icon renders electronic-application icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    electronic-application.svg
  </svg>
</div>
`;

exports[`Icon renders file-text icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    file-text.svg
  </svg>
</div>
`;

exports[`Icon renders group icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    group.svg
  </svg>
</div>
`;

exports[`Icon renders handshake icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    handshake.svg
  </svg>
</div>
`;

exports[`Icon renders home icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    home.svg
  </svg>
</div>
`;

exports[`Icon renders illustration-frame-rounded icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    illustration-frame-rounded.svg
  </svg>
</div>
`;

exports[`Icon renders info icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    info.svg
  </svg>
</div>
`;

exports[`Icon renders library icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    library.svg
  </svg>
</div>
`;

exports[`Icon renders logout icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    logout.svg
  </svg>
</div>
`;

exports[`Icon renders mail icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    mail.svg
  </svg>
</div>
`;

exports[`Icon renders mail-x icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    mail-x.svg
  </svg>
</div>
`;

exports[`Icon renders old-check-white icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    check-white.svg
  </svg>
</div>
`;

exports[`Icon renders padlock icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    padlock.svg
  </svg>
</div>
`;

exports[`Icon renders people icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    people.svg
  </svg>
</div>
`;

exports[`Icon renders plus icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    plus.svg
  </svg>
</div>
`;

exports[`Icon renders products icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    products.svg
  </svg>
</div>
`;

exports[`Icon renders pru-logo icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    logo.svg
  </svg>
</div>
`;

exports[`Icon renders refresh icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    refresh.svg
  </svg>
</div>
`;

exports[`Icon renders search icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    search.svg
  </svg>
</div>
`;

exports[`Icon renders trash icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    trash.svg
  </svg>
</div>
`;

exports[`Icon renders trash-2 icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    trash-2.svg
  </svg>
</div>
`;

exports[`Icon renders user-plus icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    user-plus.svg
  </svg>
</div>
`;

exports[`Icon renders x icon 1`] = `
.c0 {
  display: inline-block;
  cursor: inherit;
}

.c0 > svg {
  height: 100%;
  display: block;
}

<div
  class="c0"
  data-testid="icon"
>
  <svg>
    x.svg
  </svg>
</div>
`;
